docs_src/design/adr/device-service/0012-DeviceService-Filters.md

```diff
@@ -0,0 +1,83 @@
+# Device Service Filters
+
+## Status
+
+### Proposed
+
+- design for Hanoi
+- implementation for Ireland
+
+## Context
+
+In EdgeX today, sensor/device data collected can be "filtered" by [application services](../../../microservices/application/ApplicationServices.md) before being exported or sent to some [north side](../../../general/Definitions.md#south-and-north-side) application or system. Built-in application service functions (available through the app services SDK) allow EdgeX event/reading objects to be filtered by device name or by value descriptor type.  That is, event/readings can be filtered by:
+
+- which device sent the event/reading (as determined by the device name in the event/readings)
+- the category (such as temperature or humidity) of data as determined by the value descriptor.
+
+*With value descriptor's deprecation, what will the app function SDK offer for filtering?*
+
+### Filter by device name and value descriptor nearer collection
+
+While application service filters help to limit the amount of data sent north, those events and readings are still collected, stored, processed by the EdgeX micro services, potentially resulting in wasted resources processing unneeded data.
+
+Therefore, there is a desire to help filter data at the device service level - thereby limiting the amount of sensor data that is captured, stored and processed as well as sent north.
+
+In the first stage of device service filtering, it is desired that the filter by device name and value descriptor be offered in all device services (via the SDK).  The filtering of event/readings should be accomplished just before the event/readings are sent to core data or (or put on the message bus in future implementations where DS message application services).  See Design below for more details.
+
+### Sharing filter functions
+
+If one were to explore the filtering functions in the app functions SDK [filter.go](https://github.com/edgexfoundry/app-functions-sdk-go/blob/master/pkg/transforms/filter.go) (both FilterByDeviceName and FilterByValueDescriptor), the filters operate on a simple event model object.  Ideally, since both app services and device services share the event model (from go-mod-core-contracts), it would be the desire to share filter functions to some degree between SDKs.

```

Comment:
To do this we would have to move the transforms into a new go-mod-<name> repo.

Reply:
Yes - we would need some sort of shared library.  Crawl, walk, run.  First implementation may not do this, but I'd like to think filtering functions on event/readings behave the same in either location (DS or AS).